### PR TITLE
Add '--sort' option for sorting diff command output

### PR DESCRIPTION
**Pull request by Ape**
This is a solution to #797. The commit is quite trivial and should be easy to review.

Previously, on `borg diff`, the output always had first the modifications, then additions, and finally removals. Output may be easier to follow if the various kinds of changes are interleaved. This commit is a simple solution that first collects the output lines and sorts them by file path before printing. This new behavior is optional and disabled by default. It can be enabled with `--sort` command line option.

This option will be especially useful after the planned multi-threading changes arrive. Multi-threading may shuffle the archive order of files making diff output hard to follow without sorting.

### CODE REVIEW

**Comment by enkore** [comment]
If you add a small test case for sorted output you can stop CodeCov from complaining (and get a test case for sorted output for free ;-)

File: borg/archiver.py [hunk 1]
```diff
@@ -544,43 +544,56 @@ def compare_items(path, item1, item2, hardlink_masters, deleted=False):
 
             changes = [x for x in changes if x]
             if changes:
-                print("{:<19} {}".format(' '.join(changes), remove_surrogates(path)))
+                item = (remove_surrogates(path), ''.join(changes))
```
Comment (ThomasWaldmann) [inline borg/archiver.py]: above, you used a blank to join the changes, now you use an empty string?

File: borg/archiver.py [hunk 2]
```diff
@@ -544,43 +544,56 @@ def compare_items(path, item1, item2, hardlink_masters, deleted=False):
...
                     b'chunks': [],
                 }, added, hardlink_masters, deleted=True)
             for deleted in orphans_archive1.values():
-                compare_items(deleted[b'path'], deleted, {
+                compare_items(items, deleted[b'path'], deleted, {
                     b'deleted': True,
                     b'chunks': [],
                 }, hardlink_masters, deleted=True)
 
+            for item in sorted(items, key=lambda x: x[0]):
```
Comment (ThomasWaldmann) [inline borg/archiver.py]: is key=... necessary? a tuple sorts by first, second, ... element anyway.

Comment (ThomasWaldmann) [inline borg/archiver.py]: also, is a "if args.sort" missing here? if it is not sorted, you already printed the output and you don't want to print it again, sorted.

File: borg/archiver.py [hunk 3]
```diff
@@ -513,7 +513,7 @@ def compare_mode(item1, item2):
             if item1[b'mode'] != item2[b'mode']:
                 return '[{} -> {}]'.format(get_mode(item1), get_mode(item2))
 
-        def compare_items(path, item1, item2, hardlink_masters, deleted=False):
+        def compare_items(items, path, item1, item2, hardlink_masters, deleted=False):
```
Comment (ThomasWaldmann) [inline borg/archiver.py]: about names:

one would somehow expect, that items are a list of ... items - and item1 is one item and item2 is another item. but in fact, items has a different type of elements - it is a list of output data, so maybe rather call it output?

File: borg/archiver.py (hunk 2 above)
Comment (enkore) [inline borg/archiver.py]: If args.sort is False, "items" / the list of output lines isn't populated, they are printed directly (lines 549ff)

Comment (Ape) [inline borg/archiver.py]: Well `items` is empty if `args.sort` was false.

**Comment by Ape** [comment]
@ThomasWaldmann I addressed your comments in this new version.

**Comment by ThomasWaldmann** [comment]
I mentioned that there is a "if args.sort" missing. Well, you ignored that correctly - it is not needed as output list will be empty, so the code there won't do anything in that case.

**Comment by ThomasWaldmann** [comment]
OK, good change. :) Can you write a little test for it?

**Comment by Ape** [comment]
@ThomasWaldmann Added a test.
